boilerplate/tsconfig.json

```
@@ -11,7 +11,8 @@
     "noImplicitThis": true,
     "noUnusedLocals": true,
     "sourceMap": true,
-    "target": "es2015"
+    "target": "es2015",
+    "lib": ["es6"]
```

I think we should probably just make this `"lib": ["esNext"]`. I know we have a bunch of projects that use newer features.

---

@kevinvangelder Want to update this?

---

@bryanstearns Do you think this is a good idea, or should we hold off on this?

---

I'm not an expert in cutting edge Javascript but since I play one on TV, my preference is to avoid using `esNext` in real projects since its definition changes over time, and "proposed" features' definitions sometimes change before they're accepted.

I'd have no problem changing to the latest specific version, like `"lib": ["es2018"]`, though; this gives us [`{ ...rest} = x` spreading in assignments](https://2ality.com/2016/10/rest-spread-properties.html), which is the latest future-y thing I've needed recently.

---

FYI: Just checked `react-native init foo --template=typescript` -- it sets `lib` to `["es6"]`, and `target` to `"esnext"`, which doesn't make sense to me: seems to mean "please assume my Javascript engine supports all the proposed stuff, but only allow me to use es6/es2015 syntax in my code" ¯\_(ツ)_/¯

---

FYI: `react-native init foo --template=typescript` sets `target` to "es2015" and `lib` to `["es6"]`. (I removed a comment here where I'd misinterpreted what it did.)

---

@jamonholmgren I've tested this both as-is and with this change, and both are good!

```suggestion
    "lib": ["es2018"]
```